### PR TITLE
hex string DB index

### PR DESCRIPTION
**Pull request by mikiquantum**
<This is intended as a guide when describing-reviewing pull requests for go-centrifuge>
Issue Link: <Github link>

### CODE REVIEW

File: identity/did.go [hunk 1]
```diff
@@ -108,6 +108,26 @@ type DID common.Address
 // DIDLength contains the length of a DID
 const DIDLength = common.AddressLength
 
+// MarshalJSON marshals DID to json bytes.
+func (d DID) MarshalJSON() ([]byte, error) {
```
Comment (mikiquantum) [inline identity/did.go]: Added these custom marshaler, so this field appear human readable in the DB

File: identity/did.go [hunk 2]
```diff
@@ -108,6 +108,26 @@ type DID common.Address
...
+	str := "\"" + d.String() + "\""
```
Comment (vimukthi-git) [inline identity/did.go]: why we need the quotes?

Comment (mikiquantum) [reply identity/did.go]: It is the JSON standard way of serializing as string vs other types.

File: identity/did.go [hunk 3]
```diff
@@ -108,6 +108,26 @@ type DID common.Address
...
+	return []byte(str), nil
+}
+
+// UnmarshalJSON loads json bytes to DID
+func (d *DID) UnmarshalJSON(data []byte) error {
+	cleanedData := data
+	if len(data) > 2 && data[0] == '"' && data[len(data)-1] == '"' {
+		cleanedData = data[1 : len(data)-1]
+	}
```
Comment (vedhavyas) [inline identity/did.go]: I think we should use the trim prefix and suffix methods from the `strings` package

**Pull request merged**